src/extractor/edge_based_graph_factory.cpp

```diff
@@ -442,9 +445,14 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
 
                 distance += turn_penalty;
 
-                BOOST_ASSERT(m_compressed_edge_container.HasEntryForID(edge_from_u));
+                const NodeID smaller = node_u < node_v ? node_u : node_v;
+                const NodeID larger = node_u > node_v ? node_u : node_v;
+                const EdgeID smaller_to_larger_edge = m_node_based_graph->FindEdge(smaller, larger);
+                const EdgeData &smaller_to_larger_edge_data = m_node_based_graph->GetEdgeData(smaller_to_larger_edge);
+                const bool is_encoded_forwards = smaller_to_larger_edge_data.edge_id == SPECIAL_NODEID ? node_u > node_v : node_u < node_v;
```

Comment:
I think what also should work is:

- Change `CompressedEdgeContainer` to have `m_forward_edge_id_to_zipped_index_map` and `m_reverse_edge_id_to_zipped_index_map`.
- Instead of `HasZippedEntryForID` implement `IsEncodedForward()` and `IsEncodedBackward()` that return true if there is an entry in the corresponding map
- Change `GetZippedPositionForID()` to be directional: `GetZippedPositionForForwardID()` .. `GetZippedPositionForReverseID()`
- You can set `is_encoded_forward` depending on the `CompressedGeometryContainer` functions

That way the only thing we rely on is that for every edge id there is an entry, either in forward or reverse encoding, no explicit numbering assumptions.